dot/core/service.go

```diff
@@ -398,6 +400,59 @@ func (s *Service) ProcessBlockAnnounceMessage(msg network.Message) error {
 	return nil
 }
 
+// ProcessBlockRequestMessage processes a block request message, returning a block response message
+func (s *Service) ProcessBlockRequestMessage(msg network.Message) error {
+	blockRequest := msg.(*network.BlockRequestMessage)
+
+	latestBlockNum := s.blockState.LatestHeader().Number
+
+	msgStartingBlockPrefix := blockRequest.StartingBlock[0]
+	msgStartingBlockBody := blockRequest.StartingBlock[1:]
+
+	var msgStartingBlockNum *big.Int
+	var msgBlockData *types.BlockData
+	var err error
+
+	if msgStartingBlockPrefix == 1 {
+		msgStartingBlockNum = big.NewInt(0).SetBytes(msgStartingBlockBody)
```

Comment:
same here, need to use binary.LittleEndian to convert the bytes to uint64 then to big.int